lib/utils/get-element-type.js

```diff
@@ -0,0 +1,36 @@
+const {elementType, getProp, getPropValue} = require('jsx-ast-utils')
+
+/*
+Allows custom component to be mapped to an element type.
+When a default is set, all instances of the component will be mapped to the default.
+If a prop determines the type, it can be specified with `props`.
+
+For now, we only support the mapping of one prop type to an element type, rather than combinations of props.
```

Comment:
This eslint custom component mapping approach was inspired by, `jsx-a11y`. however, `jsx-a11y` only supports 1:1 mapping whereas we want to be able to associate a prop with a component.

(read more about [jsx-a11y custom component mapping](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y#usage))